feat(tabs): added tabs demos (#3914)
patternfly/patternfly: src/patternfly/demos/Tabs/examples/Tabs.md

                __tabs-item--aria-label="Installation"
                __tabs-item--text="Installation"
                __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-installation-panel"')}}
              {{> __tabs-item
                __tabs-item--id="limits"
                __tabs-item--aria-label="Limits"
                __tabs-item--text="Limits"
                __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-limits-panel"')}}
              {{> __tabs-item
                __tabs-item--id="cluster-installation"
                __tabs-item--aria-label="Cluster installation"
                __tabs-item--text="Cluster installation"
                __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-cluster-installation-panel"')}}
              {{> __tabs-item
                __tabs-item--id="subscription-support"
                __tabs-item--aria-label="Subscription and support"
                __tabs-item--text="Subscription and support"
                __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-subscription-support-panel"')}}
            {{/tabs-list}}
          {{/tabs}}
        {{/sidebar-panel}}

>         {{#> sidebar-content sidebar-content--modifier="pf-u-p-lg"}}

Reviewer: It would be great if we can avoid using a utility class in the demo. Maybe the sidebar should have padding like the drawer?

Reviewer: Technically you could use a page section. The only thing about page sections is that they expect to be in a flex layout as flex children. Putting them in `.pf-c-page__main-group` is a way to introduce them somewhere other than a direct child of `.pf-c-page__main`. Though I think the sidebar should probably have padding and that seems like a suitable approach.

          {{#> tab-content tab-content--IsActive="true" tab-content--attribute=(concat 'aria-labelledby="' tab-content--id '-installation-link" id="' tab-content--id '-installation-panel"')}}
            {{#> content}}
              <h2>Installation content</h2>

>               <p>This document is a guide for preparing a new AWS account for use with OpenShift. It will help prepare an account tot crea a single cluster and provide insight for adjustments which may be needed for additional cluster.</p>

Reviewer: ```suggestion
              <p>This document is a guide for preparing a new AWS account for use with OpenShift. It will help prepare an account to create a single cluster and provide insight for adjustments which may be needed for additional cluster.</p>
```

---
id: 'Tabs'

> section: demos

Reviewer: so it shows up as the demos tab on the component, not in the demos section in the vertical nav

```suggestion
section: components
```

---

## Examples

### Open tabs

```hbs isFullscreen
{{#> tabs--page-wrapper tabs--page-wrapper--id="open-tabs-example"}}
  {{> page-template-breadcrumb page-template-breadcrumbs--IsPods="true"}}
  {{#> page-main-section page-main-section--IsLimitWidth="true" page-main-section--modifier="pf-m-light"}}
    {{> tabs--pod-page-title}}
  {{/page-main-section}}

>   {{#> page-main-tabs page-main-tabs--IsLimitWidth="true" page-main-tabs--modifier="pf-m-light pf-m-tabs"}}

Reviewer: Do you need that? Looks like you set `background-color` on `.pf-c-page__main-tabs`. If not, it's in other spots in the demos, too.

```suggestion
  {{#> page-main-tabs page-main-tabs--IsLimitWidth="true" page-main-tabs--modifier="pf-m-tabs"}}
```

            {{> tabs--panel-header tabs--panel-header--title="Node 2" tabs--panel-header--sub-title='<a href="#">siemur/test-space</a>'}}
            {{#> drawer-body drawer-body--modifier="pf-m-no-padding"}}
              {{> primary-detail-panel-tabs primary-detail-panel-tabs--modifier="pf-m-box pf-m-fill"}}
            {{/drawer-body}}
            {{#> drawer-body primary-detail-template--id=(concat tabs--page-wrapper--id '-tabs')}}
              {{> primary-detail-panel-tab-content progress--modifier="pf-m-sm" primary-detail-panel-tab-content--HasLabels="true"}}
            {{/drawer-body}}
          {{/drawer-panel}}
        {{/drawer-main}}
      {{/drawer}}
    {{/tab-content}}
    {{#> tab-content tab-content--attribute=(concat 'aria-labelledby="' tab-content--id '-node-connectors-link" id="' tab-content--id '-node-connectors-panel"')}}
      {{#> content}}
        <p>Node connectors content</p>
      {{/content}}
    {{/tab-content}}
  {{/page-main-section}}
{{/tabs--page-wrapper}}
```

### Tables and tabs, auto width tabs
```hbs isFullscreen
{{#> tabs--page-wrapper tabs--page-wrapper--id="table-tabs-w-secondary-example"}}

>   {{#> page-main-section page-main-section--IsLimitWidth="true" page-main-section--modifier="pf-m-light"}}

Reviewer: ```suggestion
  {{#> page-main-section page-main-section--modifier="pf-m-light"}}
```

    {{#> title title--modifier="pf-m-2xl pf-u-mt-md"}}
      Nodes
    {{/title}}
  {{/page-main-section}}

>   {{#> page-main-tabs page-main-tabs--IsLimitWidth="true" page-main-tabs--modifier="pf-m-light pf-m-tabs"}}

Reviewer: You've got `.pf-m-light.pf-m-tabs` on a bunch of these - do you need those? Looks like you set the background color on `.pf-c-page__main-tabs` and that modifier isn't defined in the CSS anywhere.

>   {{#> page-main-tabs page-main-tabs--IsLimitWidth="true" page-main-tabs--modifier="pf-m-light pf-m-tabs"}}

Reviewer: ```suggestion
  {{#> page-main-tabs page-main-tabs--modifier="pf-m-light pf-m-tabs"}}
```

    {{#> tabs tabs--id=(concat tabs--page-wrapper--id '-tabs') tabs--modifier="pf-m-inset-md pf-m-inset-lg-on-xl"}}
      {{#> tabs-list}}
        {{> __tabs-item
          __tabs-item--current="true"
          __tabs-item--id="nodes"
          __tabs-item--aria-label="Nodes"
          __tabs-item--text="Nodes"
          __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-nodes-panel"')}}
        {{> __tabs-item
          __tabs-item--id="node-connectors"
          __tabs-item--aria-label="node-connectors"
          __tabs-item--text="Node connectors"
          __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-node-connectors-panel"')}}
      {{/tabs-list}}
    {{/tabs}}
  {{/page-main-tabs}}

>   {{#> page-main-section page-main-section--IsLimitWidth="true" page-main-section--modifier="pf-m-no-padding pf-m-light" tab-content--id=(concat tabs--page-wrapper--id '-tabs')}}

Reviewer: ```suggestion
  {{#> page-main-section page-main-section--modifier="pf-m-no-padding pf-m-light" tab-content--id=(concat tabs--page-wrapper--id '-tabs')}}
```

Reviewer: This will match the demo above it. The drawer area should be full width.

      {{#> content}}
        <p>Node connectors content</p>
      {{/content}}
    {{/tab-content}}
  {{/page-main-section}}
{{/tabs--page-wrapper}}
```

### Modal tabs
```hbs isFullscreen
{{#> example-wrapper example-wrapper--id="modal-tabs-example"}}
  {{#> modal}}
    {{#> backdrop}}
      {{#> bullseye}}
        {{#> modal-box modal-box--modifier="pf-m-sm" modal-box--attribute=(concat 'aria-labelledby="' example-wrapper--id '-modal-title" aria-describedby="' example-wrapper--id '-modal-description"')}}
          {{#> button button--modifier="pf-m-plain" button--attribute='aria-label="Close dialog"'}}
            <i class="fas fa-times" aria-hidden="true"></i>
          {{/button}}
          {{#> modal-box-header}}
            {{#> modal-box-title modal-box-title--attribute=(concat 'id="' example-wrapper--id '-modal-title"')}}
              Modal title
            {{/modal-box-title}}
          {{/modal-box-header}}
          {{#> modal-box-body modal-box-body--attribute=(concat 'id="' example-wrapper--id '-modal-description"')}}

>             {{#> grid grid--modifier="pf-m-gutter"}}

Reviewer: Why are we using grid here but flex in https://patternfly-pr-3914.surge.sh/components/tabs/html-demos/nested-tabs/?

          __tabs-item--text="Get started"
          __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-get-started-panel"')}}
        {{> __tabs-item
          __tabs-item--id="knowledge"
          __tabs-item--aria-label="Knowledge"
          __tabs-item--text="Knowledge"
          __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-knowledge-panel"')}}
        {{> __tabs-item
          __tabs-item--id="support"
          __tabs-item--aria-label="Support"
          __tabs-item--text="Support"
          __tabs-item--attribute=(concat 'aria-controls="' tabs--id '-support-panel"')}}
      {{/tabs-list}}
    {{/tabs}}
  {{/page-main-tabs}}
  {{#> page-main-section page-main-section--IsLimitWidth="true" tab-content--id=(concat tabs--page-wrapper--id '-tabs') page-main-section--modifier="pf-m-light"}}
    {{#> tab-content tab-content--attribute=(concat 'aria-labelledby="' tab-content--id '-new-link" id="' tab-content--id '-new-panel"')}}
      What's new content
    {{/tab-content}}
    {{#> tab-content tab-content--IsActive="true" tab-content--attribute=(concat 'aria-labelledby="' tab-content--id '-get-started-link" id="' tab-content--id '-get-started-panel"')}}

>       {{#> grid grid--modifier="pf-m-gutter"}}

Reviewer: Also grid but flex in another example that looks like the same layout as far as I can tell.
